### PR TITLE
client/Txn: set status on elided `EndTransaction`

### LINKED ISSUE #2399: `client.Txn`: set status for elided `EndTransaction`
`client.Txn.send` may silently drop `EndTransaction` calls if the transaction is read-only. The client may reasonably expect `Txn.Proto.Status` to be in a terminal state (committed or aborted) after sending an `EndTransaction` call, so when `Txn.send` drops an `EndTransaction`, it should update its in-memory `Transaction.Status` (to committed if there were no errors, or aborted if there were).

This would let us remove the setting of `Status` in sql/server.go.

From #2396.

### PR DESCRIPTION
**Pull request by tamird**
Closes #2399.

### CODE REVIEW

File: client/txn.go [hunk 1]
```diff
@@ -406,12 +406,21 @@ func (txn *Txn) send(calls ...proto.Call) error {
...
 
-	if haveEndTxn && !needEndTxn {
+	if elideEndTxn {
 		calls = calls[:lastIndex]
 	}
 
-	return txn.db.send(calls...)
+	err := txn.db.send(calls...)
+	if elideEndTxn && err == nil {
+		if endTxnRequest.Commit {
```
Comment (spencerkimball) [inline client/txn.go]: Add a comment here specifying that all close-out status in returned transaction must be set so it's clear to future maintainers that if something besides status is expected to be set, it's meant to be handled here for the case of elided end txn.

    // Manually set transaction status (and any other information typically set on
    // commit or abort based on success or failure of the batched calls.

**Comment by spencerkimball** [comment]
LGTM. Nice to see that nasty SQL-level case pushed further down.

File: client/txn.go (hunk 1 above)
Comment (tamird) [inline client/txn.go]: added a comment and some tests; sadly had to touch the test infrastructure more than I would have liked.

**Comment by mrtracy** [comment]
LGTM

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**